File: 2465-number-of-distinct-averages/2465-number-of-distinct-averages.rb
# @param {Integer[]} nums
# @return {Integer}
def distinct_averages(nums)
  nums.sort!
  avgs = Set.new
  while nums.length > 0
    min, max = nums.shift, nums.pop
    avgs.add((min+max)/2.0)
  end
  return avgs.size
end

File: 2465-number-of-distinct-averages/2465-number-of-distinct-averages.py
class Solution:
  def distinctAverages(self, nums: List[int]) -> int:
    nums.sort()
    avgs = set()
    l, r = 0, len(nums)-1
    while (l < r):
      avgs.add((nums[l]+nums[r])/2)
      l += 1
      r -= 1
    return len(avgs)